src/worldcereal/job.py

```diff
@@ -167,21 +185,52 @@ def generate_map(
     if job_options is not None:
         JOB_OPTIONS.update(job_options)
 
+    # Compile filename of final product
+    proc_level = "raw" if postprocess_parameters.enable is False else "cleaned"
+    filename = f"{product_type.value}-{proc_level}"
+
+    # Execute the job
     job = classes.execute_batch(
-        outputfile=output_path,
         out_format=out_format,
         job_options=JOB_OPTIONS,
         title="WorldCereal [generate_map] job",
         description="Job that performs end-to-end WorldCereal inference",
+        filename_prefix=filename,
     )
 
-    asset = job.get_results().get_assets()[0]
+    # Get job results
+    job_result = job.get_results()
+
+    # Get the products
+    assets = job_result.get_assets()
+    products = {}
+    for asset in assets:
+        name = asset.name.split("_")[0]
+        prod_type = (
+            WorldCerealProductType.CROPLAND
+            if name.split("-")[0] == "cropland"
```

Comment:
this looks a bit odd an not so future proof (when more products would be added, this thing is gonna get ugly). Maybe we should in the filename use the exact match of the name of the product, such as `CROPLAND` or `CROPTYPE`. This way, you can directly construct the `WorldCerealProductType` from the name and avoid `if-else` statements.

Reply:
fixed in commit. The WorldCerealProductType is now constructed from the asset name without the if - else statement